src/domain/tokens/utils.ts

```diff
@@ -1,18 +1,97 @@
-import { BigNumber } from "ethers";
-import { getVisibleTokens } from "../../config/Tokens";
+import { BigNumber, ethers } from "ethers";
 import {
-  bigNumberify,
-  expandDecimals,
+  adjustForDecimals,
+  DUST_BNB,
   getFeeBasisPoints,
-  getTokenInfo,
+  MARKET,
   MINT_BURN_FEE_BASIS_POINTS,
   PRECISION,
   TAX_BASIS_POINTS,
-  USDG_DECIMALS,
   USD_DECIMALS,
   USDG_ADDRESS,
+  USDG_DECIMALS,
 } from "../../lib/legacy";
+import { NATIVE_TOKENS_MAP, TOKENS, TOKENS_BY_SYMBOL_MAP, TOKENS_MAP, WRAPPED_TOKENS_MAP } from "../../config/tokens";
+import { getExplorerUrl } from "../../config/chains";
 import { InfoTokens, Token, TokenInfo } from "./types";
+import { bigNumberify, expandDecimals } from "../../lib/numbers";
+
+const { AddressZero } = ethers.constants;
+
+export function getWrappedToken(chainId: number) {
+  return WRAPPED_TOKENS_MAP[chainId];
+}
+
+export function getNativeToken(chainId: number) {
+  return NATIVE_TOKENS_MAP[chainId];
+}
+
+export function getTokens(chainId: number) {
+  return TOKENS[chainId];
+}
+
+export function isValidToken(chainId: number, address: string) {
+  if (!TOKENS_MAP[chainId]) {
+    throw new Error(`Incorrect chainId ${chainId}`);
+  }
+  return address in TOKENS_MAP[chainId];
+}
+
+export function getToken(chainId: number, address: string) {
+  if (!TOKENS_MAP[chainId]) {
+    throw new Error(`Incorrect chainId ${chainId}`);
+  }
+  if (!TOKENS_MAP[chainId][address]) {
+    throw new Error(`Incorrect address "${address}" for chainId ${chainId}`);
+  }
+  return TOKENS_MAP[chainId][address];
+}
+
+export function getTokenUrl(chainId: number, address: string) {
+  if (!address) {
+    return getExplorerUrl(chainId);
+  }
+  return getExplorerUrl(chainId) + "token/" + address;
+}
+
+export function getTokenBySymbol(chainId: number, symbol: string) {
+  const token = TOKENS_BY_SYMBOL_MAP[chainId][symbol];
+  if (!token) {
+    throw new Error(`Incorrect symbol "${symbol}" for chainId ${chainId}`);
+  }
+  return token;
+}
+
+export function getWhitelistedTokens(chainId: number) {
+  return TOKENS[chainId].filter((token) => token.symbol !== "USDG");
+}
+
+export function getVisibleTokens(chainId: number) {
+  return getWhitelistedTokens(chainId).filter((token) => !token.isWrapped && !token.isTempHidden);
+}
+
+export function getUsd(
```

Comment:
not sure if `getUsd` should be inside tokens utils. logically it related to not tokens only?

Reply:
It depends on tokenAddress and InfoTokens, uses token structure inside, looks like this is the function that applies specifically to tokens :)

I don't really like that it depends on orderOption as well, as it could be made more abstract so that it only depends on triggerPrice. But overall it looks like a function from the token domain. 

Also, in practice I think it's convenient to have functions like getTokenAmountFromUsd and getUsd (from token amount) in one place

Reply:
> I don't really like that it depends on orderOption as well

agree on that. if you think it makes sense it in this PR – let's do it. otherwise we can do it later